mysql-test/t/acl_mdl_scalability.test

```diff
@@ -0,0 +1,255 @@
+   WHERE state = 'debug sync point: acl_s_lock' AND
+         info = 'SHOW CREATE USER root@localhost';
+--source include/wait_condition.inc
+
+--echo # Start 5000 connection will go into ACL cache lock waiting state
+--let $i = 1
+while ($i <= 5000) {
+ #--echo "starting new connection $i that will blocked"
+ --exec_in_background $MYSQL -usatya -phello -h 127.0.0.1 -P 13000 test -e "SELECT 1" 2>&1 > /dev/null
+ if ($i == 1500) {
+   --connection con_flush
+   --send FLUSH PRIVILEGES;
+ }
+ --inc $i
+}
+
+--echo # Connection default: Wait for those 5000 new connections to reach ACL cache lock waiting state
+--connection default
+--let $wait_timeout = 600
+LET $wait_condition=
+  SELECT COUNT(*) = 5000 FROM information_schema.processlist
+    WHERE USER = 'unauthenticated user' AND STATE = 'Waiting for acl cache lock';
+--source include/wait_condition.inc
+
+--echo # Start another 500 connections and stop them in just before deadlock check iteration
+--let $i = 201
+while ($i <= 700) {
+  --echo "starting new connection $i that will blocked in Deadlock check
+  --connection con$i
+  SET DEBUG_SYNC = 'now WAIT_FOR s_locked NO_CLEAR_EVENT';
+  SET DEBUG_SYNC = 'mdl_acquire_lock_wait SIGNAL before_dead_lock WAIT_FOR start_dead_lock_check NO_CLEAR_EVENT';
+  SET DEBUG_SYNC = 'acl_mdl_dead_lock SIGNAL dead_locked WAIT_FOR resume_wait_graph NO_CLEAR_EVENT';
+  --send SHOW CREATE USER root@localhost
+  --inc $i
+}
+
+--echo # Connection default: Make 500 threads stop just before deadlock check
+--connection default
+SET DEBUG_SYNC = 'now WAIT_FOR before_dead_lock NO_CLEAR_EVENT';
+
+--echo # Now signal waiting threads to start deadlock checks
+SET DEBUG_SYNC = 'now SIGNAL start_dead_lock_check';
+
+--echo # Waiting for 500 threads to reach before deadlock check state
+--let $wait_timeout = 60
+LET $wait_condition=
+  SELECT COUNT(*) = 500 FROM information_schema.processlist
+    WHERE STATE = 'debug sync point: mdl_acquire_lock_wait'
+    AND info = 'SHOW CREATE USER root@localhost';
+#--sleep 30
+
+--echo # Signal 500 threads to start deadlocks checks
+SET DEBUG_SYNC = 'now SIGNAL resume_wait_graph';
+--echo # New let the initial 200 readers who took the SHARED lock to resume
+--echo # They will not be able to release the lock/ticket to internal rwlock
+--echo # taken as SHARED by Deadlock check threads. These threads need exclusive
+--echo # mode to remove ticket from granted list
+SET DEBUG_SYNC = 'now SIGNAL new_xlock';
+
+--echo # database freezes now and new connections are blocked
+--echo # and finally we hit max_connections limit
+--echo # Freeze start..  Also observe high cpu usage.. mysql server is doing deadlock checks
+--echo # try connecting from external client
+--echo # ../runtime_output_directory/mysql -usatya -phello -h 127.0.0.1 -P 13000 test -e "SELECT 1"
+--echo # and to measure time to connect use
+--echo # time ../runtime_output_directory/mysql -usatya -phello -h 127.0.0.1 -P 13000 test -e "SELECT 1"
```

Comment:
```suggestion
--echo # time ../runtime_output_directory/mysql -usatya -phello -h 127.0.0.1 -P $MASTER_MYPORT test -e "SELECT 1"
```

Reply:
Likewise

Reply:
on changing the comment to $MY_PORT, it is tricky.

1. --replace_result doesn't work on --echo
2.  The reason behind this message is to help user when debugging. (possibly without result file. That is how I measured before adding the code for freeze time). On second thought, since we measure hang time, we can get rid of this prints altogether.